src/snowflake/sqlalchemy/snowdialect.py

```diff
@@ -669,7 +672,10 @@ def get_columns(self, connection, table_name, schema=None, **kw):
         if schema_columns is None:
             # Too many results, fall back to only query about single table
             return self._get_table_columns(connection, table_name, schema, **kw)
-        return schema_columns[self.normalize_name(table_name)]
+        normalized_table_name = self.normalize_name(table_name)
+        if normalized_table_name not in schema_columns:
+            raise sa_exc.NoSuchTableError()
+        return schema_columns[normalized_table_name]
```

Comment:
could you help check and do us a favor to update if there're other similar usages of the `index` operator to get value from a dict which will trigger unexpected `KeyError` in `snowdialect.py`?

Reply:
I did a review when I made this change for exactly this, but I just did another one, and here are all of the places I found that might lead to a key error. Let me know if any of these should be fixed and what error we should throw instead:

- L209: `host` is assumed to be in `opts` in `create_connect_args`
  - Not sure what we would want to throw here. 
- L329-336: A few keys are assumed to be returned by the query in `_get_schema_primary_keys`
  - This would be an internal error if it happened, so not sure a better error would be needed.
- L361-370: Same thing here in `_get_schema_unique_constraints`
- L401-423: Same thing again in `_get_schema_foreign_keys`
- L559: Dependency on internal API between `_get_schema_primary_keys` and `_get_schema_columns`. Shouldn't occur without a bug in the code.
- L655: Same here between `_get_schema_primary_keys` and `_get_table_columns`.
- L747: Another instance where we are depending on the results from the `connection.execute` call on `SHOW VIEWS...`. Regardless, exception is handled and ignored, so no KeyError would be shown to user.
- L769-670: Same thing here, but no catch of exceptions, so a KeyError would be shown to user. That said, it would be an internal bug if it happened, so not something we need to customize the error for.
- L838-839: Another instance of dependency on the result of a query, which would cause an internal error if it weren't true.

Reply:
thanks for collecting the usage. for the first one, I think this would be addressed by the PR: https://github.com/snowflakedb/snowflake-sqlalchemy/pull/387.

for the remaining ones, I concur with you that those should not require extra error handling.
again thanks for the efforsts